### PR TITLE
Use upstream libp2p

### PR DESCRIPTION
**Pull request by rishflab**
todo: remove ping

### CODE REVIEW

**Comment by rishflab** [comment]
the rendezvous `unit` test failed locally. There were no failures in CI and im pretty sure it would of run. Investigating... may just be a one off failure

**Comment by thomaseizinger** [comment]
It is not passing CI either but hanging forever, I will have a look!

File: swap/src/network/cbor_request_response.rs [hunk 1]
```diff
@@ -59,7 +55,7 @@ where
     where
         T: AsyncRead + Unpin + Send,
     {
-        let message = upgrade::read_one(io, BUF_SIZE)
+        let message = upgrade::read_length_prefixed(io, BUF_SIZE)
             .await
             .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
```
Comment (thomaseizinger) [inline swap/src/network/cbor_request_response.rs:60]: Also get rid of this.

File: swap/src/network/json_pull_codec.rs [hunk 2]
```diff
@@ -55,7 +55,7 @@ where
     where
         T: AsyncRead + Unpin + Send,
     {
-        let message = upgrade::read_one(io, BUF_SIZE)
+        let message = upgrade::read_length_prefixed(io, BUF_SIZE)
             .await
             .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
```
Comment (thomaseizinger) [inline swap/src/network/json_pull_codec.rs:60]: And here.

File: swap/src/cli.rs [hunk 3]
```diff
@@ -138,14 +138,14 @@ mod tests {
         #[behaviour(ignore)]
         registered: bool,
     }
-
-    impl NetworkBehaviourEventProcess<libp2p::rendezvous::Event> for StaticQuoteAsbBehaviour {
-        fn inject_event(&mut self, event: libp2p::rendezvous::Event) {
-            if let libp2p::rendezvous::Event::Registered { .. } = event {
+    impl NetworkBehaviourEventProcess<libp2p::rendezvous::client::Event> for StaticQuoteAsbBehaviour {
```
Comment (thomaseizinger) [inline swap/src/cli.rs:141]: Doing `use libp2p::rendezvous;` would go  along way in this file as well.

File: swap/src/network/cbor_request_response.rs (hunk 1 above)
Comment (thomaseizinger) [reply swap/src/network/cbor_request_response.rs:60]: Can we not use `?` here?

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thanks! Couple of comments :)

File: swap/src/cli.rs [hunk 4]
```diff
@@ -59,10 +59,9 @@ mod tests {
...
-        let mut rendezvous_node = new_swarm(|_, identity| RendezvousPointBehaviour {
-            rendezvous: libp2p::rendezvous::Rendezvous::new(
-                identity,
-                libp2p::rendezvous::Config::default(),
+        let mut rendezvous_node = new_swarm(|_, _| RendezvousPointBehaviour {
+            rendezvous: libp2p::rendezvous::server::Behaviour::new(
+                libp2p::rendezvous::server::Config::default(),
             ),
             ping: Default::default(),
         });
```
[the remark this answers is not in the document]
Comment (rishflab) [reply swap/src/cli.rs:67]: its only constructed once and in a test. not sure if worth it

**Pull request merged**